### PR TITLE
Feat/DEVSU-2247 Add Search Reports by Variant Feature

### PR DESCRIPTION
**Pull request by bnguyen-bcgsc**
- Add search view to search for reports with specific variant
- Create new reports view for search result
- Validate input variant value
- Add new icon and search menu item to sidebar
- Add support for svg import to typescript config
- Code linting

### CODE REVIEW

File: app/views/SearchView/index.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  TextField,
+  Typography,
+  InputAdornment,
+  IconButton,
+} from '@mui/material';
+import SearchIcon from '@mui/icons-material/Search';
+import ArrowCircleRightIcon from '@mui/icons-material/ArrowCircleRight';
+import React, { useCallback, useEffect, useState } from 'react';
+
+const MIN_WORD_LENGTH = 3;
```
Comment (elewis2) [inline app/views/SearchView/index.tsx:13]: there are genes with 2-letter names - reduce this to 2

**Review by elewis2 — APPROVED** [review]

**Review by elewis2 — DISMISSED** [review]

**Review by kttkjl — CHANGES_REQUESTED** [review]
Some comments on styling on ticket as well

File: app/utils/searchReportsColumns.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import startCase from '@/utils/startCase';
+
+function searchReportsColumns(report, analyst, reviewer, bioinformatician) {
```
[the remark this answers is not in the document]
Comment (bnguyen-bcgsc) [reply app/utils/searchReportsColumns.js:3]: Yea I think we might reuse it with future advanced search features (when searching for reports using other fields than key variant)

File: app/views/SearchView/index.tsx [hunk 3]
```diff
@@ -116,9 +116,6 @@ const SearchView = () => {
           />
         </div>
         <div className="search__threshold-input" onKeyUp={(event) => event.keyCode === ENTER_KEYCODE && handleSubmit()}>
```
Comment (kttkjl) [inline app/views/SearchView/index.tsx:118]: keycode is deprecated, we should use key instead

**Review by kttkjl — APPROVED** [review]

**Pull request merged**